src/main/java/org/trigon/commands/CameraPositionCalculationCommand.java

```diff
@@ -0,0 +1,115 @@
+package org.trigon.commands;
+
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import org.littletonrobotics.junction.networktables.LoggedNetworkNumber;
+
+import java.util.Optional;
+import java.util.function.DoubleConsumer;
+import java.util.function.Supplier;
+
+public class CameraPositionCalculationCommand extends Command {
+    private static final LoggedNetworkNumber ROTATION_SPEED = new LoggedNetworkNumber("CameraPositionCalculationCommand/RotationSpeed", 1);
+    private static final LoggedNetworkNumber ROTATION_RATE_LIMIT = new LoggedNetworkNumber("CameraPositionCalculationCommand/RotationRateLimit", 1);
+
+    private final Supplier<Optional<Pose2d>> cameraPose;
+    private final Rotation2d minimumRotationDistance;
+    private final DoubleConsumer rotateRobot;
+    private final Supplier<Rotation2d> robotHeading;
+
+    private SlewRateLimiter rotationSlewRateLimiter;
+    private boolean hasSetFirstCameraPose = false;
+    private Pose2d startingCameraPose = null;
+    private Rotation2d StartingCameraPoseHeading;
+    private Transform2d robotToCamera = null;
+
+    public CameraPositionCalculationCommand(Supplier<Optional<Pose2d>> cameraPose, Rotation2d minimumRotationDistance, DoubleConsumer rotateRobot, Supplier<Rotation2d> robotHeading, SubsystemBase requirement) {
```

Comment:
Supplier<Optional<Pose2d>>: why optional 
minimumRotationDistance: no
Supplier<Rotation2d> robotHeading: no need, perhaps you're implementing this wrong

add a param called for the angle of the camera on the robot. in the end, you should log and print a transform2d with that heading, and rotate the translation fed into it by that variable